docs/tests/secret-mysql.md

```diff
@@ -0,0 +1,69 @@
+# MySQL Secret Benchmark `mysql_secret`
+
+This benchmark will test the dynamic generation of MySQL credentials.
+
+## Test Parameters
+
+### DB Connection Configuration `db_connection`
+
+- `name` `(string: "benchmark-mysql")` - Name for this database connection. This is specified as part of the URL.
+- `plugin_name` `(string: "mysql-database-plugin")` - Specifies the name of the plugin to use for this connection.
+- `plugin_version` `(string: "")` - Specifies the semantic version of the plugin to use for this connection.
+- `verify_connection` `(bool: true)` - Specifies if the connection is verified during initial configuration. Defaults to true.
+- `allowed_roles` `(list: ["benchmark_role"])` - List of the roles allowed to use this connection.
+- `root_rotation_statements` `(list: [])` - Specifies the database statements to be executed to rotate the root user's credentials.
+- `password_policy` `(string: "")` - The name of the password policy to use when generating passwords for this database. If not specified, this will use a default policy defined as: 20 characters with at least 1 uppercase, 1 lowercase, 1 number, and 1 dash character.
+- `connection_url` `(string: <required>)` - Specifies the MySQL DSN. This field can be templated and supports passing the username and password parameters in the following format `{{field_name}}`. A templated connection URL is required when using root credential rotation.
+- `disable_escaping` `(boolean: false)` - Turns off the escaping of special characters inside of the username and password fields. See the [databases secrets engine docs](https://developer.hashicorp.com/vault/docs/secrets/databases#disable-character-escaping) for more information. Defaults to `false`.
+- `max_open_connections` `(int: 4)` - Specifies the maximum number of open connections to the database.
```

Comment:
just out of curiosity have we tested this? if we don't set `max_open_connections` will it get sent to the api as the zero value (0)?

Reply:
No it doesn't look like it.  Turns out even if we _explicitly_ set it to 0 (not sure why someone would, feel like that would defeat the purpose) it gets set to 4 also https://github.com/hashicorp/vault/blob/476bec104e2519b9d3454669cda6eba012a8a44c/plugins/database/mysql/connection_producer.go#L78-L79.